File: cpp/Keyboard.cpp
// Keyboard.cpp (Hybrid Inheritance)
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "PeripheralDevice.cpp"
using namespace std;

class Keyboard : public Komponen, public PeripheralDevice {
private:
    bool mechanical;
    string layoutType;
    bool hasBacklight;
    
public:
    Keyboard() {}
    
    Keyboard(bool mechanical, string layoutType, bool hasBacklight,
           string interfaceType, bool isWireless,
           string merk, string nama) : 
           Komponen(merk, nama), PeripheralDevice(interfaceType, isWireless) {
        this->mechanical = mechanical;
        this->layoutType = layoutType;
        this->hasBacklight = hasBacklight;
    }
    
    void setMechanical(bool mechanical) {
        this->mechanical = mechanical;
    }
    
    void setLayoutType(string layoutType) {
        this->layoutType = layoutType;
    }
    
    void setHasBacklight(bool hasBacklight) {
        this->hasBacklight = hasBacklight;
    }
    
    bool getMechanical() {
        return this->mechanical;
    }
    
    string getLayoutType() {
        return this->layoutType;
    }
    
    bool getHasBacklight() {
        return this->hasBacklight;
    }
    
    ~Keyboard() {}
};

File: python/Keyboard.py
from Komponen import Komponen
from PeripheralDevice import PeripheralDevice
class Keyboard(Komponen, PeripheralDevice) :
    def __init__(self, mechanical, layoutType, hasBackLight, merk, nama, interfaceType, isWireless):
        Komponen.__init__(self, merk, nama)
        PeripheralDevice.__init__(self, interfaceType, isWireless)
        
        self.__mechanical = mechanical
        self.__layoutType = layoutType
        self.__hasBackLight = hasBackLight
        
    def get_mechanical(self):
        return self.__mechanical
    def get_layoutType(self):
        return self.__layoutType
    def get_hasBackLight(self):
        return self.__hasBackLight
    
    # Setter methods
    def set_mechanical(self, mechanical):
        self.__mechanical = mechanical
    def set_layoutType(self, layoutType):
        self.__layoutType = layoutType
    def set_hasBackLight(self, hasBackLight):
        self.__hasBackLight = hasBackLight
        